modules/network_synchronizer/networked_controller.cpp

```diff
@@ -0,0 +1,1313 @@
+		if (peers[i].active == false) {
+			// Nothing to do on this peer.
+			continue;
+		}
+
+		bool force_dispatch_batch = false;
+
+		peers[i].collect_timer += p_delta;
+		if (
+				is_epoch_important ||
+				peers[i].collect_timer >= (node->get_doll_state_collect_rate() / peers[i].update_rate_factor)) {
+			// Resets the timer.
+			peers[i].collect_timer -= node->get_doll_state_collect_rate() / peers[i].update_rate_factor;
+			// Needed because is possible to force send the state update.
+			peers[i].collect_timer = MAX(0.0, peers[i].collect_timer);
+
+			if (epoch_state_collected == false) {
+				epoch_state_data.begin_write();
+				epoch_state_data.add_int(epoch, DataBuffer::COMPRESSION_LEVEL_1);
+				node->call("collect_epoch_data", &epoch_state_data);
+				epoch_state_data.dry();
+				epoch_state_collected = true;
+			}
+
+			if (is_epoch_important) {
+				force_dispatch_batch = true;
+				node->rpc_id(
+						peers[i].peer,
+						"_rpc_doll_send_epoch",
+						epoch_state_data.get_buffer().get_bytes());
+			} else {
+				// Store this into epoch batch.
+				// TODO count buffer size.
+				if (unlikely(
+							// If the packet is more than 255 it can't be sent
+							// into a batch.
+							epoch_state_data.get_buffer().get_bytes().size() > UINT8_MAX ||
+							// If added this into the batch would cause the IP
+							// fragmentation. So just send the data in two different
+							// packets.
+							peers[i].batch_size + epoch_state_data.get_buffer().get_bytes().size() + 1 >= 1350)) {
+					if (epoch_state_data.get_buffer().get_bytes().size() > UINT8_MAX) {
+						ERR_PRINT("The status update is too big, try to staty under 255 byte per update, so the batch system can be used.");
+					} else {
+						WARN_PRINT("The amount of data collected per batch is more than 1350 bytes. Please make sure the `doll_sync_timer_rate` is not so big, so the batch can be correctly created. Batch size: " + itos(peers[i].batch_size) + " - Epochs into the batch: " + itos(peers[i].epoch_batch.size()));
+					}
+					force_dispatch_batch = true;
+					node->rpc_unreliable_id(
+							peers[i].peer,
+							"_rpc_doll_send_epoch",
+							epoch_state_data.get_buffer().get_bytes());
+				} else {
+					peers[i].batch_size += 1 + epoch_state_data.get_buffer().get_bytes().size();
+					peers[i].epoch_batch.push_back(epoch_state_data.get_buffer().get_bytes());
+				}
+			}
+		}
+
+		peers[i].sync_timer += p_delta;
+		if (
+				peers[i].epoch_batch.size() != 0 && // Has something.
+				(
+						force_dispatch_batch ||
+						peers[i].sync_timer >= node->get_doll_state_sync_rate())) {
+			peers[i].sync_timer -= node->get_doll_state_sync_rate();
+
+			// Prepare the batch
+			Vector<uint8_t> data;
+			// TODO Resize the array here.
+			for (uint32_t x = 0; x < peers[i].epoch_batch.size(); x += 1) {
+				ERR_CONTINUE_MSG(peers[i].epoch_batch[x].size() > 256, "It's not allowed to send more then 256 bytes per status. This status is dropped.");
```

Comment:
Typo: "to send more then 256 bytes" -> "to send more than 256 bytes"

Reply:
Done ty